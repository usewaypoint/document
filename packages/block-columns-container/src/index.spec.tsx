import React from 'react';

import { render } from '@testing-library/react';

import { ColumnsContainer } from '.';

describe('block-columns-container', () => {
  it('renders with default values', () => {
    expect(render(<ColumnsContainer />).asFragment()).toMatchSnapshot();
  });

  describe('columnsCount 2', () => {
    it('renders column children', () => {
      const columns = [<>bread</>, <>tomato</>, <>lettuce</>];
      expect(render(<ColumnsContainer props={{ columnsCount: 2 }} columns={columns} />).asFragment()).toMatchSnapshot();
    });

    it('uses padding correctly', () => {
      const columns = [<>bread</>, <>tomato</>, <>lettuce</>];
      expect(
        render(
          <ColumnsContainer
            props={{
              columnsGap: 12,
              columnsCount: 2,
            }}
            columns={columns}
          />
        ).asFragment()
      ).toMatchSnapshot();
    });
  });

  describe('columnsCount 3', () => {
    it('renders column children', () => {
      const columns = [<>bread</>, <>tomato</>, <>lettuce</>];
      expect(render(<ColumnsContainer props={{ columnsCount: 3 }} columns={columns} />).asFragment()).toMatchSnapshot();
    });

    it('uses padding correctly', () => {
      const columns = [<>bread</>, <>tomato</>, <>lettuce</>];
      expect(
        render(
          <ColumnsContainer
            props={{
              columnsGap: 12,
              columnsCount: 3,
            }}
            columns={columns}
          />
        ).asFragment()
      ).toMatchSnapshot();
    });
  });
});
