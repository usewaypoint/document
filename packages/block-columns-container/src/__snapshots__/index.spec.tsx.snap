// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block-columns-container columnsCount 2 renders column children 1`] = `
<DocumentFragment>
  <div>
    <table
      align="center"
      border="0"
      cellpadding="0"
      style="table-layout: fixed; border-collapse: collapse;"
      width="100%"
    >
      <tbody
        style="width: 100%;"
      >
        <tr
          style="width: 100%;"
        >
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 0px; padding-right: 0px;"
          >
            bread
          </td>
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 0px; padding-right: 0px;"
          >
            tomato
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`block-columns-container columnsCount 2 uses padding correctly 1`] = `
<DocumentFragment>
  <div>
    <table
      align="center"
      border="0"
      cellpadding="0"
      style="table-layout: fixed; border-collapse: collapse;"
      width="100%"
    >
      <tbody
        style="width: 100%;"
      >
        <tr
          style="width: 100%;"
        >
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 0px; padding-right: 6px;"
          >
            bread
          </td>
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 6px; padding-right: 0px;"
          >
            tomato
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`block-columns-container columnsCount 3 renders column children 1`] = `
<DocumentFragment>
  <div>
    <table
      align="center"
      border="0"
      cellpadding="0"
      style="table-layout: fixed; border-collapse: collapse;"
      width="100%"
    >
      <tbody
        style="width: 100%;"
      >
        <tr
          style="width: 100%;"
        >
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 0px; padding-right: 0px;"
          >
            bread
          </td>
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 0px; padding-right: 0px;"
          >
            tomato
          </td>
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 0px; padding-right: 0px;"
          >
            lettuce
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`block-columns-container columnsCount 3 uses padding correctly 1`] = `
<DocumentFragment>
  <div>
    <table
      align="center"
      border="0"
      cellpadding="0"
      style="table-layout: fixed; border-collapse: collapse;"
      width="100%"
    >
      <tbody
        style="width: 100%;"
      >
        <tr
          style="width: 100%;"
        >
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 0px; padding-right: 8px;"
          >
            bread
          </td>
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 4px; padding-right: 4px;"
          >
            tomato
          </td>
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 8px; padding-right: 0px;"
          >
            lettuce
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`block-columns-container renders with default values 1`] = `
<DocumentFragment>
  <div>
    <table
      align="center"
      border="0"
      cellpadding="0"
      style="table-layout: fixed; border-collapse: collapse;"
      width="100%"
    >
      <tbody
        style="width: 100%;"
      >
        <tr
          style="width: 100%;"
        >
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 0px; padding-right: 0px;"
          />
          <td
            style="box-sizing: content-box; vertical-align: middle; padding-left: 0px; padding-right: 0px;"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
